core/WASM_core.md

```diff
@@ -0,0 +1,352 @@
+# Summary
+
+This proposal intends to outline the shape of the interfaces and implementation work necessary
+to make Core WASM-compilable.
+
+# Motivation
+
+A few problems a WASM Core Can solve, in no particular order:
+
+* Allow Go and Java to be ported to run on top of Core. *Potentially* without the downsides of CGo
+  and JNI. More on that later.
+* No longer any need to ship compiled Core binaries of various architectures with our existing SDKs.
```

Comment:
While there may no longer be a "need", there's surely still considerable performance improvement to doing so (at least for .NET and Python). For any existing core-based SDK the tradeoffs of course should be discussed because I don't think we want to maintain two approaches in any SDK.